### PR TITLE
Win32

### PR DESCRIPTION
**Pull request by rpavlik**
I picked up your work and did a little more with it to try to be able to use it with SolveSpace. Here's what I have so far, which appears to at least minimally work though the scaling needs tweaking on the Solvespace side.

### CODE REVIEW

File: src/usbdev.c [hunk 1]
```diff
@@ -81,6 +81,12 @@ static int usbdev_getled(struct spndev* dev);
 static inline void usbdev_parsebuttons(const struct spndev* dev, union spndev_event* evt, const unsigned char* report);
 static inline void checkrange(const struct spndev* dev, const int val);
 
+#ifdef _WIN32
+#define VID_PID_FORMAT_STR "%04x:%04x"
+#else
+#define VID_PID_FORMAT_STR "%#0.4hx:%#0.4hx"
```
Comment (ruevs) [inline src/usbdev.c:96]: Crash? It dos not crash for me on Win32 with `"%#0.4hx:%#0.4hx"` (VS 2019).

**Comment by ruevs** [comment]
Thanks a lot! I merged it.
I'm curious - how did you stumble upon this?

File: src/usbdev.c (hunk 1 above)
Comment (rpavlik) [reply src/usbdev.c:96]: The crash might ahve been from when I was still trying to build it with msys2 and mingw. The ifdef is thus almost certainly too broad.

**Comment by rpavlik** [comment]
I was digging around to see if there was an easy solution to quickly bolt in spacemouse support to SolveSpace for Windows 64-bit builds, since I got a spacemouse recently and I'm currently on Windows most of the time.

**Pull request merged**